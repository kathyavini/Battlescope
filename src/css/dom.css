* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

html {
    min-width: 100%;
    min-height: 100%;
    overflow-x: hidden;
}

body {
    width: 100vw;
    height: 100%;
    position: relative;
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;
}

h1 {
    font-size: clamp(2rem, 2vw + 1rem, 2.5rem);
}

/* .announce-panel h3 {
    border: 1px solid greenyellow;
    width: 100px;
    height: 50px;
} */



.announce-panel {
    /* border: 1px solid white; */
    min-height: 5rem;
    width: 500px;
    height: 100px;
    color: var(--accent5);
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    opacity: 0;
    transition: opacity 500ms 500ms color 0ms 250ms;
    font-size: clamp(1rem, 2vw + 1rem, 1.5rem);
}

.announce-panel.hit {
    font-size: clamp(1.25rem, 2vw + 1rem, 2rem);
    color: var(--accent1);
}

.game-announce.win {
    width: 100%;
    font-size: clamp(1.25rem, 2vw + 1rem, 2.5rem);
    color: var(--accent);
    opacity: 1;
    margin-right: 0;
    text-align: center;
}

.announce-panel.visible {
    opacity: 1;
    transition: opacity 500ms;
    /* delay is second number gosh */
}

.play-area {
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    align-items: center;
    row-gap: 50px;
    column-gap: 200px;
    text-align: center;
    --hit-gradient: radial-gradient(transparent 30%, var(--dark3) 100%);
}

.play-area section {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    flex: 0;
}

.board {
    width: 90vw;
    max-width: 450px;
    min-width: 300px;
    height: 90vw;
    max-height: 450px;
    min-height: 300px;
    display: grid;
    grid-template-rows: repeat(10, 1fr);
    grid-template-columns: repeat(10, 1fr);
    border: 1px solid var(--white);
    background-color: var(--dark3);
}

.square {
    border: 1px solid var(--dark4);
}

.square:hover {
    background-color: var(--accent2);
    opacity: 0.5;
}

.square:active {
    filter: brightness(1.2)
}

.own-board .square.ship {
/* .square.ship { */
    background-color: var(--accent);
}

.square.miss {
    background-color: var(--accent5);
}

/* .square.ship.hit {
    background-color: var(--accent1);
} */

/* .square.ship.sunk {
    filter: brightness(.4);
} */

.sunk-fleet {
    width: 600px;
    height: 200px;
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    row-gap: 5px;
}

.ship-render {
    flex-shrink: none;
    flex-grow: none;
    margin: 20px;
    display: flex;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
    background-color: var(--light2);
    border: 1px solid var(--light3);
}

.ship-part {
    background-color: transparent;
    width: 40px;
    height: 40px;
    /* border: 1px solid var(--accent7); */
    /* background-color: var(--accent1); */
    /* filter: brightness(.7); */
}

.square.ship.hit {
    background-image: var(--hit-gradient);
}

.square.ship.tagged {
    background-size: cover;
    background-repeat: no-repeat;
}


.square.ship.tagged.p {
    background-image: url("../svg/crab-sweet.svg");
}

.square.ship.hit.p.tagged,
.ship-render.p {
    background-image: var(--hit-gradient), url("../svg/crab-sketch.svg");
}

.square.ship.tagged.s {
    background-image: url("../svg/narwhal-sweet.svg");
}

.square.ship.hit.s.tagged,
.ship-render.s {
    background-image: var(--hit-gradient), url("../svg/narwhal-front.svg");
}

.square.ship.tagged.d{
    background-image: url("../svg/blowfish-side.svg");
}

.square.ship.hit.d.tagged,
.ship-render.d  {
    background-image: var(--hit-gradient), url("../svg/blowfish-front.svg");
}


.square.ship.tagged.b {
    background-image: url("../svg/dolphin-sweet.svg");
}

.square.ship.hit.b.tagged,
.ship-render.b {
    background-image: var(--hit-gradient), url("../svg/dolphin-cheerful.svg");
    background-position: right;
}

.ship-render.b {
    background-position: center;
}

.square.ship.tagged.c {
    background-image: url("../svg/whale-sweet.svg");
    background-size: contain;

}

.square.ship.hit.c.tagged,
.ship-render.c {
    background-image: var(--hit-gradient), url("../svg/whale-front.svg");
    background-size: contain;
}
