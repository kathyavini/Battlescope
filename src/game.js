import {
  createContainers,
  makeAnnouncements,
  clickListener,
  renderBoard,
  renderTurnScreen,
  renderStartScreen,
} from './dom';
import { renderShipScreen } from './dragDrop';
import createPlayer from './player';
import createGameboard from './gameboard';
import { subscribe, publish } from './pubsub';

export function newGame() {
  createContainers();

  const player1 = createPlayer();
  const player2 = createPlayer();

  const board1 = createGameboard();
  const board2 = createGameboard();

  player1.assignEnemyGameboard(board2);
  player2.assignEnemyGameboard(board1);

  board1.placeAllShipsRandomly();
  board2.placeAllShipsRandomly();

  // Event listeners to track game events
  makeAnnouncements();

  // Event listeners to select type of game
  const [singlePlayerButton, twoPlayerButton] = renderStartScreen();

  singlePlayerButton.addEventListener('click', (ev) => {
    ev.preventDefault();
    document.querySelector('.start-screen').style.display = 'none';

    playerVsAILoop({ player1, player2, board1, board2 });
  });

  twoPlayerButton.addEventListener('click', (ev) => {
    ev.preventDefault();
    document.querySelector('.start-screen').style.display = 'none';

    twoPlayerGameLoop({ player1, player2, board1, board2 });
  });
}

function playerVsAILoop({ player1, player2, board1, board2 }) {
  renderShipScreen(board1); // interesting that doing something with this promise is not necessary

  renderBoard(board1.boardArray, 'own');
  renderBoard(board2.boardArray, 'enemy');

  // Make board clickable to human player
  clickListener(player1, 'enemy');

  subscribe('squareAttacked', humanAttack);

  function humanAttack([player, target]) {
    publish('targetChange', 'enemy');

    player.makeAttack([target[0], target[1]]);

    renderBoard(board2.boardArray, 'enemy');

    if (board2.isFleetSunk()) {
      publish('fleetSunk', ['Computer', 'Human']);
      return;
    }

    setTimeout(() => {
      publish('targetChange', 'own');

      player2.makeAttack(player2.aiSmartPlay());
      renderBoard(board1.boardArray, 'own');

      if (board1.isFleetSunk()) {
        publish('fleetSunk', ['Human', 'Computer']);
        return;
      }

      setTimeout(() => {}, 1500);
    }, 1000);
  }
}

async function twoPlayerGameLoop({ player1, player2, board1, board2 }) {
  
  await renderTurnScreen('PLAYER 1').then(console.log)
  await renderShipScreen(board1).then(console.log);

  await renderTurnScreen('PLAYER 2').then(console.log);
  await renderShipScreen(board2).then(console.log);

  // Game loop

}
