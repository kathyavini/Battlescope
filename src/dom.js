import './css/colors-and-fonts.css';
import './css/dom.css';
import './css/mobile.css';
import { createNewElement } from './utils';
import { subscribe, publish } from './pubsub';

export function createContainers() {
  const body = document.querySelector('body');
  const boardsContainer = createNewElement('main', ['play-area']);
  const enemyContainer = createNewElement('section', ['enemy']);
  const ownContainer = createNewElement('section', ['own']);

  enemyContainer.append(
    createNewElement('h2', ['subtitle', 'enemy-title'], 'Enemy Waters'),
    createNewElement('h3', ['announce-panel']),
    createNewElement('div', ['board', 'enemy-board']),
    createNewElement('div', ['sunk-fleet'])
  );

  ownContainer.append(
    createNewElement('h2', ['subtitle', 'own-title'], 'Friendly Waters'),
    createNewElement('h3', ['announce-panel']),
    createNewElement('div', ['board', 'own-board']),
    createNewElement('div', ['sunk-fleet'])
  );
  boardsContainer.append(enemyContainer, ownContainer);

  body.append(
    createNewElement('h1', ['title'], 'Battlescope'),
    createNewElement('h3', ['game-announce']),
    boardsContainer
  );

  /* Make squares */
  const boards = document.querySelectorAll('.board');
  for (const board of boards) {
    for (let i = 0; i < 10; i++) {
      for (let j = 0; j < 10; j++) {
        const div = createNewElement('div', ['square'], null, {
          'data-pos': `${i}${j}`,
        });
        board.appendChild(div);
      }
    }
  }
}

export function renderBoardExperimental(gameboard, section) {
  // Drag and drop event listeners
  let thisDragElement;

  document.addEventListener('dragstart', (ev) => {
    thisDragElement = ev.target;
  });

  document.addEventListener('dragenter', (ev) => {
    if (ev.target.className === 'square') {
      ev.target.style.backgroundColor = 'purple';
    }
  });

  document.addEventListener('dragleave', (ev) => {
    if (ev.target.className === 'square') {
      ev.target.style.backgroundColor = '';
    }
  });

  document.addEventListener('dragover', (ev) => {
    // MDN says this is necessary to allow drop
    ev.preventDefault();
  });

  document.addEventListener('drop', (ev) => {
    ev.preventDefault();

    if (ev.target.className === 'square') {
      ev.target.style.backgroundColor = '';

      const coords = [+ev.target.dataset.pos[0], +ev.target.dataset.pos[1]];

      const type = thisDragElement.dataset.type;
      const orientation = thisDragElement.dataset.orientation;

      console.log(coords, type, orientation);

      try {
        gameboard.placeShip(type, coords, orientation);
        console.log('this was a legal placement');

        thisDragElement.parentNode.removeChild(thisDragElement);
        ev.target.appendChild(thisDragElement);

        thisDragElement.dataset.pos = ev.target.dataset.pos;

        clearShipFromBoard(thisDragElement, gameboard);
        updateFleet(thisDragElement, gameboard);
        gameboard.placeShip(type, coords, orientation)

      } catch (error) {
        console.log(error);
        console.log('this was not a legal placement');
      }

      // console.log(thisDragElement.dataset);
    }
  });

  for (let i = 0; i < 10; i++) {
    for (let j = 0; j < 10; j++) {
      const value = gameboard.boardArray[i][j];
      const boardSquare = document.querySelector(
        `.${section}-board [data-pos="${i}${j}"]`
      );

      if (!value) {
        continue;
      } else if (value[1] === '1') {
        const [ship] = gameboard.fleet.filter((x) => x.type[0] == value[0]);

        const draggable = renderDraggableShip(ship);

        boardSquare.appendChild(draggable);

        draggable.addEventListener('click', (ev) => {
          ev.stopPropagation();

          let newOrientation;

          if (ev.target.dataset.orientation === 'vertical') {
            newOrientation = 'horizontal';
          } else {
            newOrientation = 'vertical';
          }

          const coords = [
            +draggable.parentNode.dataset.pos[0],
            +draggable.parentNode.dataset.pos[1],
          ];

          const type = draggable.dataset.type;

          console.log(type, coords, newOrientation);

          // I do wish I had written a function "is legal placement" instead of throwing an error within place ship... this makes it a lot harder
          try {
            gameboard.placeShip(type, coords, newOrientation);

            console.log('this was a legal placement');

            ev.target.classList.toggle('rotated');
            ev.target.dataset.orientation = newOrientation;

            clearShipFromBoard(draggable, gameboard);
            updateFleet(draggable, gameboard);

            gameboard.placeShip(type, coords, newOrientation);

          } catch (error) {
            console.log(error);
            console.log('this was not a legal placement');
          }
        });
      }
    }
  }
}

// 
function clearShipFromBoard(updatedShip, gameboard) {
  console.log('clearing');

  let newBoard = gameboard.boardArray.map((x) => [...x]);
  console.log(newBoard);

  for (let i = 0; i < 10; i++) {
    for (let j = 0; j < 10; j++) {
      let value = newBoard[i][j];

      if (!value) {
        continue;
      } else if (value[0] === updatedShip.dataset.type[0]) {
        console.log('Cleared value: ' + value);
        value = null;
      }
    }
  }
  console.log(newBoard);
  // added a setter to gameboard for this
  gameboard.boardArray = newBoard;
}

// For orientation; actually not needed until the end, right?
function updateFleet(updatedShip, gameboard) {
  let newFleet = gameboard.fleet;
  const targetShip = newFleet.filter(
    (x) => (x.type = updatedShip.dataset.type)
  );

  console.log(targetShip);
  console.log(gameboard.fleet);
}

function renderDraggableShip(ship) {
  const shipRender = createNewElement(
    'div',
    ['draggable-ship', `${ship.type[0]}`],
    null,
    {
      draggable: 'true',
      'data-orientation': ship.orientation,
      'data-length': `${ship.length}`,
      'data-type': ship.type,
    }
  );

  if (ship.orientation === 'vertical') {
    shipRender.classList.add('rotated');
  }

  for (let i = 0; i < ship.length; i++) {
    shipRender.appendChild(createNewElement('div', ['ship-part']));
  }

  return shipRender;
}

export function renderBoard(board, section) {
  for (let i = 0; i < 10; i++) {
    for (let j = 0; j < 10; j++) {
      const value = board[i][j];
      const boardSquare = document.querySelector(
        `.${section}-board [data-pos="${i}${j}"]`
      );
      if (!value) {
        continue;
      } else if (value === 'hit' || value === 'miss') {
        boardSquare.classList.add(value);
      } else if (value === 'sunk') {
        boardSquare.classList.add(value);
        boardSquare.classList.add('hit');
      } else {
        boardSquare.classList.add(value);
        boardSquare.classList.add('ship');

        // THE ANIMALS!
        boardSquare.classList.add(`${value[0]}`);

        if (value[0] === 'b' && value[1] === '2') {
          boardSquare.classList.add('tagged');
        } else if (value[0] === 'c' && value[1] === '3') {
          boardSquare.classList.add('tagged');
        } else if ((value[0] === 's' || value[0] === 'd') && value[1] === '2') {
          boardSquare.classList.add('tagged');
        } else if (value[0] === 'p' && value[1] === '1') {
          boardSquare.classList.add('tagged');
        }
      }
    }
  }
}

function clickAttack(ev, player) {
  console.log('click attack');
  publish('squareAttacked', [player, ev.target.dataset.pos]);

  ev.target.style.pointerEvents = 'none';

  ev.target.parentElement.style.pointerEvents = 'none';

  // Controls time until next human click can be registered
  setTimeout(() => {
    ev.target.parentElement.style.pointerEvents = 'initial';
  }, 15);
}

export function clickListener(player, section) {
  const targetContainer = document.querySelector(`section.${section}`);

  const targetSquares = targetContainer.querySelectorAll('.square');
  targetSquares.forEach((square) => {
    square.addEventListener('click', (ev) => {
      clickAttack(ev, player);
    });
  });
}

export function makeAnnouncements(player, section) {
  const targetContainer = document.querySelector(`section.${section}`);

  let panel;
  let sunkFleet;
  let visibleContainer;
  const gamePanel = document.querySelector('.game-announce');
  const boardsContainer = document.querySelector('main');
  const subtitles = document.querySelectorAll('.subtitle');

  subscribe('targetChange', changePanel);

  function changePanel(target) {
    panel = document.querySelector(`.${target} .announce-panel`);
    sunkFleet = document.querySelector(`.${target} .sunk-fleet`);
  }

  subscribe('boardChange', changeView);
  const containers = document.querySelectorAll('.play-area section');

  function changeView(target) {
    for (const container of containers) {
      container.classList.remove('visible');
    }
    visibleContainer = document.querySelector(`.${target}`);
    visibleContainer.classList.add('visible');
  }

  const gamePanels = document.querySelectorAll('.announce-panel');

  for (const gamePanel of gamePanels) {
    gamePanel.addEventListener('transitionend', () => {
      gamePanel.classList.remove('visible');
      gamePanel.classList.remove('hit');
    });
  }

  subscribe('hit', announceHit);
  subscribe('miss', announceMiss);
  subscribe('shipSunk', announceSunkShip);
  subscribe('fleetSunk', announceWin);
  subscribe('shipSunk', renderSunkShip);

  function announceHit([row, column]) {
    panel.classList.add('visible');
    panel.classList.add('hit');
    panel.textContent = `Something's been spotted!`;
  }

  function announceMiss() {
    panel.classList.add('visible');
    panel.textContent = `Nothing!`;
  }

  function announceSunkShip([hitShip, [row, column]]) {
    panel.classList.add('visible');
    const creature = shipMapping[`${hitShip.type}`];
    panel.textContent = `It's a ${creature}!!`;
  }

  function announceWin([loser, winner]) {
    gamePanel.classList.add('win');
    gamePanel.textContent = `${loser}'s sea creatures have been found! ${winner} wins!`;
    boardsContainer.style.pointerEvents = 'none';
    for (const gamePanel of gamePanels) {
      gamePanel.style.display = 'none';
    }
  }

  function renderSunkShip([hitShip, [row, column]]) {
    const shipRender = createNewElement('div', [
      'ship-render',
      `${hitShip.type[0]}`,
    ]);

    for (let i = 0; i < hitShip.length; i++) {
      shipRender.appendChild(createNewElement('div', ['ship-part']));
    }

    sunkFleet.appendChild(shipRender);
  }
}

/* For themeing */
const shipMapping = {
  carrier: 'octopus',
  battleship: 'pufferfish',
  destroyer: 'goldfish',
  submarine: 'seahorse',
  'patrol boat': 'betta fish',
};

export function renderTurnScreen() {
  const boardsContainer = document.querySelector('main');
  console.log(boardsContainer);

  const turnScreen = createNewElement('div', ['turn-screen']);
  turnScreen.append(
    createNewElement(
      'h2',
      ['turn-instructions'],
      'Please pass the device to the next player. Hit ready when device is passed'
    ),
    createNewElement('button', ['ready-button'], 'Ready')
  );

  boardsContainer.append(turnScreen);
}
