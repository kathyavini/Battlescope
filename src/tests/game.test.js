/**
 * @jest-environment jsdom
 */

import { screen } from '@testing-library/dom';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { newGame } from '../game';

const { player1, player2, board1, board2 } = newGame();

test('Initial page load snapshot', () => {
  const body = document.querySelector('body');
  expect(body).toMatchSnapshot();
});

test('1 Player button loads ship placement screen', () => {
  const onePlayerBtn = screen.getByText('1-Player');
  userEvent.click(onePlayerBtn);

  const shipPlacementInstructions = screen.getByText('Hide your wildlife');

  expect(shipPlacementInstructions).toBeInTheDocument();
});

test('Ship placement board contains 5 ships', () => {
  const draggableShips = document.querySelectorAll('.draggable-ship');
  expect(draggableShips.length).toBe(5);
});

test('Ships are placed according to underlying gameboard', () => {
    const testShip = document.querySelector('.draggable-ship');
    const [row, col] = testShip.parentElement.dataset.pos;
    const boardValue = board1.boardArray[row][col];
    expect(boardValue).not.toBeNull()
});

test('Randomize button changes location of ship elements - only tests one square; rerun test if it fails', () => {
    const testShip1 = document.querySelector('.draggable-ship');
    const [row1, col1] = testShip1.parentElement.dataset.pos;

    const randomizeBtn = screen.getByText('Randomize');
    userEvent.click(randomizeBtn);

    const testShip2 = document.querySelector('.draggable-ship');
    const [row2, col2] = testShip2.parentElement.dataset.pos;

    // Technically it could coincidentally be true
    expect(row1 === row2 && col1 === col2).toBe(false);
})

test('Randomize button changes underlying gameboard', () => {
    const initialBoard = board1.boardArray;

    const randomizeBtn = screen.getByText('Randomize');
    userEvent.click(randomizeBtn);

    const newBoard = board1.boardArray;
    expect(initialBoard).not.toEqual(newBoard);
})

test("Start button doesn't alter underlying gameboard", () => {
    const initialBoard = board1.boardArray;
    
    const startBtn = screen.getByText('Start');
    userEvent.click(startBtn);

    const newBoard = board1.boardArray;
    expect(initialBoard).toEqual(newBoard);
})