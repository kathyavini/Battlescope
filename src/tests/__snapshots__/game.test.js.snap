// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial page load snapshot 1`] = `
<body>
  <h1
    class="title"
  >
    Battlescope
  </h1>
  <h3
    class="game-announce"
  />
  <main
    class="play-area"
  >
    <section
      class="enemy"
    >
      <h2
        class="subtitle enemy-title"
      >
        Opponent Ocean
      </h2>
      <h3
        class="announce-panel"
      />
      <div
        class="board-fleet"
      >
        <div
          class="board enemy-board"
        >
          <div
            class="square"
            data-pos="00"
          />
          <div
            class="square"
            data-pos="01"
          />
          <div
            class="square"
            data-pos="02"
          />
          <div
            class="square"
            data-pos="03"
          />
          <div
            class="square"
            data-pos="04"
          />
          <div
            class="square"
            data-pos="05"
          />
          <div
            class="square"
            data-pos="06"
          />
          <div
            class="square"
            data-pos="07"
          />
          <div
            class="square"
            data-pos="08"
          />
          <div
            class="square"
            data-pos="09"
          />
          <div
            class="square"
            data-pos="10"
          />
          <div
            class="square"
            data-pos="11"
          />
          <div
            class="square"
            data-pos="12"
          />
          <div
            class="square"
            data-pos="13"
          />
          <div
            class="square"
            data-pos="14"
          />
          <div
            class="square"
            data-pos="15"
          />
          <div
            class="square"
            data-pos="16"
          />
          <div
            class="square"
            data-pos="17"
          />
          <div
            class="square"
            data-pos="18"
          />
          <div
            class="square"
            data-pos="19"
          />
          <div
            class="square"
            data-pos="20"
          />
          <div
            class="square"
            data-pos="21"
          />
          <div
            class="square"
            data-pos="22"
          />
          <div
            class="square"
            data-pos="23"
          />
          <div
            class="square"
            data-pos="24"
          />
          <div
            class="square"
            data-pos="25"
          />
          <div
            class="square"
            data-pos="26"
          />
          <div
            class="square"
            data-pos="27"
          />
          <div
            class="square"
            data-pos="28"
          />
          <div
            class="square"
            data-pos="29"
          />
          <div
            class="square"
            data-pos="30"
          />
          <div
            class="square"
            data-pos="31"
          />
          <div
            class="square"
            data-pos="32"
          />
          <div
            class="square"
            data-pos="33"
          />
          <div
            class="square"
            data-pos="34"
          />
          <div
            class="square"
            data-pos="35"
          />
          <div
            class="square"
            data-pos="36"
          />
          <div
            class="square"
            data-pos="37"
          />
          <div
            class="square"
            data-pos="38"
          />
          <div
            class="square"
            data-pos="39"
          />
          <div
            class="square"
            data-pos="40"
          />
          <div
            class="square"
            data-pos="41"
          />
          <div
            class="square"
            data-pos="42"
          />
          <div
            class="square"
            data-pos="43"
          />
          <div
            class="square"
            data-pos="44"
          />
          <div
            class="square"
            data-pos="45"
          />
          <div
            class="square"
            data-pos="46"
          />
          <div
            class="square"
            data-pos="47"
          />
          <div
            class="square"
            data-pos="48"
          />
          <div
            class="square"
            data-pos="49"
          />
          <div
            class="square"
            data-pos="50"
          />
          <div
            class="square"
            data-pos="51"
          />
          <div
            class="square"
            data-pos="52"
          />
          <div
            class="square"
            data-pos="53"
          />
          <div
            class="square"
            data-pos="54"
          />
          <div
            class="square"
            data-pos="55"
          />
          <div
            class="square"
            data-pos="56"
          />
          <div
            class="square"
            data-pos="57"
          />
          <div
            class="square"
            data-pos="58"
          />
          <div
            class="square"
            data-pos="59"
          />
          <div
            class="square"
            data-pos="60"
          />
          <div
            class="square"
            data-pos="61"
          />
          <div
            class="square"
            data-pos="62"
          />
          <div
            class="square"
            data-pos="63"
          />
          <div
            class="square"
            data-pos="64"
          />
          <div
            class="square"
            data-pos="65"
          />
          <div
            class="square"
            data-pos="66"
          />
          <div
            class="square"
            data-pos="67"
          />
          <div
            class="square"
            data-pos="68"
          />
          <div
            class="square"
            data-pos="69"
          />
          <div
            class="square"
            data-pos="70"
          />
          <div
            class="square"
            data-pos="71"
          />
          <div
            class="square"
            data-pos="72"
          />
          <div
            class="square"
            data-pos="73"
          />
          <div
            class="square"
            data-pos="74"
          />
          <div
            class="square"
            data-pos="75"
          />
          <div
            class="square"
            data-pos="76"
          />
          <div
            class="square"
            data-pos="77"
          />
          <div
            class="square"
            data-pos="78"
          />
          <div
            class="square"
            data-pos="79"
          />
          <div
            class="square"
            data-pos="80"
          />
          <div
            class="square"
            data-pos="81"
          />
          <div
            class="square"
            data-pos="82"
          />
          <div
            class="square"
            data-pos="83"
          />
          <div
            class="square"
            data-pos="84"
          />
          <div
            class="square"
            data-pos="85"
          />
          <div
            class="square"
            data-pos="86"
          />
          <div
            class="square"
            data-pos="87"
          />
          <div
            class="square"
            data-pos="88"
          />
          <div
            class="square"
            data-pos="89"
          />
          <div
            class="square"
            data-pos="90"
          />
          <div
            class="square"
            data-pos="91"
          />
          <div
            class="square"
            data-pos="92"
          />
          <div
            class="square"
            data-pos="93"
          />
          <div
            class="square"
            data-pos="94"
          />
          <div
            class="square"
            data-pos="95"
          />
          <div
            class="square"
            data-pos="96"
          />
          <div
            class="square"
            data-pos="97"
          />
          <div
            class="square"
            data-pos="98"
          />
          <div
            class="square"
            data-pos="99"
          />
        </div>
        <div
          class="sunk-fleet"
        />
      </div>
    </section>
    <section
      class="own"
    >
      <h2
        class="subtitle own-title"
      >
        Your Ocean
      </h2>
      <h3
        class="announce-panel"
      />
      <div
        class="board-fleet"
      >
        <div
          class="board own-board"
        >
          <div
            class="square"
            data-pos="00"
          />
          <div
            class="square"
            data-pos="01"
          />
          <div
            class="square"
            data-pos="02"
          />
          <div
            class="square"
            data-pos="03"
          />
          <div
            class="square"
            data-pos="04"
          />
          <div
            class="square"
            data-pos="05"
          />
          <div
            class="square"
            data-pos="06"
          />
          <div
            class="square"
            data-pos="07"
          />
          <div
            class="square"
            data-pos="08"
          />
          <div
            class="square"
            data-pos="09"
          />
          <div
            class="square"
            data-pos="10"
          />
          <div
            class="square"
            data-pos="11"
          />
          <div
            class="square"
            data-pos="12"
          />
          <div
            class="square"
            data-pos="13"
          />
          <div
            class="square"
            data-pos="14"
          />
          <div
            class="square"
            data-pos="15"
          />
          <div
            class="square"
            data-pos="16"
          />
          <div
            class="square"
            data-pos="17"
          />
          <div
            class="square"
            data-pos="18"
          />
          <div
            class="square"
            data-pos="19"
          />
          <div
            class="square"
            data-pos="20"
          />
          <div
            class="square"
            data-pos="21"
          />
          <div
            class="square"
            data-pos="22"
          />
          <div
            class="square"
            data-pos="23"
          />
          <div
            class="square"
            data-pos="24"
          />
          <div
            class="square"
            data-pos="25"
          />
          <div
            class="square"
            data-pos="26"
          />
          <div
            class="square"
            data-pos="27"
          />
          <div
            class="square"
            data-pos="28"
          />
          <div
            class="square"
            data-pos="29"
          />
          <div
            class="square"
            data-pos="30"
          />
          <div
            class="square"
            data-pos="31"
          />
          <div
            class="square"
            data-pos="32"
          />
          <div
            class="square"
            data-pos="33"
          />
          <div
            class="square"
            data-pos="34"
          />
          <div
            class="square"
            data-pos="35"
          />
          <div
            class="square"
            data-pos="36"
          />
          <div
            class="square"
            data-pos="37"
          />
          <div
            class="square"
            data-pos="38"
          />
          <div
            class="square"
            data-pos="39"
          />
          <div
            class="square"
            data-pos="40"
          />
          <div
            class="square"
            data-pos="41"
          />
          <div
            class="square"
            data-pos="42"
          />
          <div
            class="square"
            data-pos="43"
          />
          <div
            class="square"
            data-pos="44"
          />
          <div
            class="square"
            data-pos="45"
          />
          <div
            class="square"
            data-pos="46"
          />
          <div
            class="square"
            data-pos="47"
          />
          <div
            class="square"
            data-pos="48"
          />
          <div
            class="square"
            data-pos="49"
          />
          <div
            class="square"
            data-pos="50"
          />
          <div
            class="square"
            data-pos="51"
          />
          <div
            class="square"
            data-pos="52"
          />
          <div
            class="square"
            data-pos="53"
          />
          <div
            class="square"
            data-pos="54"
          />
          <div
            class="square"
            data-pos="55"
          />
          <div
            class="square"
            data-pos="56"
          />
          <div
            class="square"
            data-pos="57"
          />
          <div
            class="square"
            data-pos="58"
          />
          <div
            class="square"
            data-pos="59"
          />
          <div
            class="square"
            data-pos="60"
          />
          <div
            class="square"
            data-pos="61"
          />
          <div
            class="square"
            data-pos="62"
          />
          <div
            class="square"
            data-pos="63"
          />
          <div
            class="square"
            data-pos="64"
          />
          <div
            class="square"
            data-pos="65"
          />
          <div
            class="square"
            data-pos="66"
          />
          <div
            class="square"
            data-pos="67"
          />
          <div
            class="square"
            data-pos="68"
          />
          <div
            class="square"
            data-pos="69"
          />
          <div
            class="square"
            data-pos="70"
          />
          <div
            class="square"
            data-pos="71"
          />
          <div
            class="square"
            data-pos="72"
          />
          <div
            class="square"
            data-pos="73"
          />
          <div
            class="square"
            data-pos="74"
          />
          <div
            class="square"
            data-pos="75"
          />
          <div
            class="square"
            data-pos="76"
          />
          <div
            class="square"
            data-pos="77"
          />
          <div
            class="square"
            data-pos="78"
          />
          <div
            class="square"
            data-pos="79"
          />
          <div
            class="square"
            data-pos="80"
          />
          <div
            class="square"
            data-pos="81"
          />
          <div
            class="square"
            data-pos="82"
          />
          <div
            class="square"
            data-pos="83"
          />
          <div
            class="square"
            data-pos="84"
          />
          <div
            class="square"
            data-pos="85"
          />
          <div
            class="square"
            data-pos="86"
          />
          <div
            class="square"
            data-pos="87"
          />
          <div
            class="square"
            data-pos="88"
          />
          <div
            class="square"
            data-pos="89"
          />
          <div
            class="square"
            data-pos="90"
          />
          <div
            class="square"
            data-pos="91"
          />
          <div
            class="square"
            data-pos="92"
          />
          <div
            class="square"
            data-pos="93"
          />
          <div
            class="square"
            data-pos="94"
          />
          <div
            class="square"
            data-pos="95"
          />
          <div
            class="square"
            data-pos="96"
          />
          <div
            class="square"
            data-pos="97"
          />
          <div
            class="square"
            data-pos="98"
          />
          <div
            class="square"
            data-pos="99"
          />
        </div>
        <div
          class="sunk-fleet"
        />
      </div>
    </section>
  </main>
  <div
    class="start-screen"
  >
    <h1
      class="start-title"
    >
      BattleSCOPE
    </h1>
    <div
      class="scope"
    />
    <h2
      class="start-subtitle"
    >
      A friendlier take on the classic game Battleship
    </h2>
    <p
      class="directions"
    >
      DIRECTIONS: Explore your opponent's ocean with your underwater scope. The first to spot all five sea creatures wins! In 2-PLAYER-MODE each turn grants five scope attempts.
    </p>
    <button
      class="single-player-button"
    >
      1-Player
    </button>
    <button
      class="two-player-button"
    >
      2-Player
    </button>
  </div>
</body>
`;
